sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/Hotspots/ExecutingSqlQueries.cs

```diff
@@ -51,20 +53,86 @@ public ExecutingSqlQueries()
+                    !(argumentList.Arguments[index].Expression.RemoveParentheses() is ExpressionSyntax interpolationArgument) ||
+                    !interpolationArgument.IsAnyKind(SyntaxKind.InterpolatedStringExpression) ||
+                    interpolationArgument.IsConstant(context.SemanticModel))
                 {
                     return false;
                 }
 
-                var onlyArgument = argumentList.Arguments[0].Expression.RemoveParentheses();
-
-                return onlyArgument.IsAnyKind(SyntaxKind.InterpolatedStringExpression) ||
-                    onlyArgument.IsConstant(context.SemanticModel);
+                var otherParameters = argumentList.Arguments.Skip(index + 1).ToList();
+                return !AllConstantsOrScalars(otherParameters, context.SemanticModel);
             };
+
+
+        protected override ExpressionSyntax GetArgumentAtIndex(InvocationContext context, int index) =>
+            context.Invocation is InvocationExpressionSyntax invocation
+                ? invocation.ArgumentList.Get(index)
+                : null;
+
+        protected override ExpressionSyntax GetSetValue(PropertyAccessContext context) =>
+            context.Expression is MemberAccessExpressionSyntax setter && setter.IsLeftSideOfAssignment()
+                ? ((AssignmentExpressionSyntax)setter.GetSelfOrTopParenthesizedExpression().Parent).Right.RemoveParentheses()
+                : null;
+
+        protected override ExpressionSyntax GetFirstArgument(ObjectCreationContext context) =>
+            context.Expression is ObjectCreationExpressionSyntax objectCreation
+                ? objectCreation.ArgumentList.Get(0)
+                : null;
+
+        protected override bool IsConcat(ExpressionSyntax argument, SemanticModel semanticModel) =>
+            IsStringMethodInvocation("Concat", argument, semanticModel) ||
+            (
+                argument.IsKind(SyntaxKind.AddExpression) &&
+                argument is BinaryExpressionSyntax concatenation &&
+                !IsConcatenationOfConstants(concatenation, semanticModel)
+            );
+
+        protected override bool IsFormat(ExpressionSyntax argument, SemanticModel semanticModel) =>
+            IsStringMethodInvocation("Format", argument, semanticModel);
+
+        protected override bool IsInterpolated(ExpressionSyntax argument) =>
+            argument.IsAnyKind(SyntaxKind.InterpolatedStringExpression);
+
+        private bool IsStringMethodInvocation(string methodName, ExpressionSyntax expression, SemanticModel semanticModel)
+        {
+            return expression is InvocationExpressionSyntax invocation &&
+                invocation.IsMethodInvocation(KnownType.System_String, methodName, semanticModel) &&
+                !AllConstantsOrScalars(invocation.ArgumentList.Arguments.ToList(), semanticModel);
+        }
+
+        private bool AllConstantsOrScalars(List<ArgumentSyntax> arguments, SemanticModel semanticModel) =>
```

Comment:
in the end I'm going to remove the whitelisting of scalar arguments because of a very good comment made by @duncanp-sonar : scalars should be security hotspots as well, as for example an integer value can be passed to a query `"SELECT TOP " + number + "FROM HugeTable"` and it needs revision as well (cc @Corniel )